README.md

```diff
@@ -137,13 +126,169 @@ matrix:
       env: TWINCAT_PRAGMALINT=1
 ```
 
-`TWINCAT_PROJECT_ROOT` may be specified here, defaulting to `$TRAVIS_BUILD_DIR`.
+`TWINCAT_PROJECT_ROOT` may be specified here, defaulting to
+`$TRAVIS_BUILD_DIR`.
+
+### python_linter.sh
+`python_linter.sh` lints given Python package(s) with the provided flake8
+arguments.
+
+#### usage:
+This script is left here for use in local tests. To test locally, simply run
+the script from the command line. If you wish to specify linter options, set
+`PYTHON_LINT_OPTIONS` beforehand. If you wish to run a Python linter in Travis,
+it is recommended to instead use `shared_configs/python-linter.yml`.
+``` bash
+PYTHON_LINT_OPTIONS="source --verbose"
+./python_linter.sh
+```
+
+`PYTHON_LINT_OPTIONS` can be set to include directory names and/or other flake8
+command line options. Any directory names will be relative to the repository's
+root directory.
+
+### shared_configs/python-linter.yml
+`python-linter.yml` examines Python code using flake8.
+
+#### usage:
+This configuration can be added to the `test` stage of a Travis build by
+importing it in your `.travis.yml`:
+``` yaml
+import:
+  - pcdshub/pcds-ci-helpers:travis/shared_configs/python-linter.yml
+```
+#### arguments:
+`PYTHON_LINT_OPTIONS` can be set to include directory names and/or other flake8
+command line options. Any directory names will be relative to the repository's
+root directory. For example:
+``` yaml
+env:
+  global:
+    - PYTHON_LINT_OPTIONS="source --verbose"
+```
+
+### shared_configs/python-tester.yml
+`python-tester.yml` runs any pytest tests it finds after installing the
+specified requirements.
+
+#### usage:
+This configuration can be added to the `test` stage of a Travis build by
+importing it in your `.travis.yml`:
+``` yaml
+import:
+  - pcdshub/pcds-ci-helpers:travis/shared_configs/python-tester.yml
+```
+#### arguments:
+Requirements files can be specified by assigning values to `REQUIREMENTS` and
+`DEV_REQUIREMENTS`. `REQUIREMENTS` defaults to `requirements.txt` and
+`DEV_REQUIREMENTS` defaults to `dev-requirements.txt`. If no requirements are
+necessary, the file can be blank.
+``` yaml
+env:
+  global:
+    - REQUIREMENTS: requirements.txt
+    - DEV_REQUIREMENTS: dev-requirements.txt
+```
+
+### shared_configs/doctr-upload.yml
+`docs-build.yml` runs through the build of the package's documentation to
+ensure it works properly.
+
+#### usage:
+This configuration can be added to the `test` stage of a Travis build by
+importing it in your `.travis.yml`:
+``` yaml
+import:
+  - pcdshub/pcds-ci-helpers:travis/shared_configs/docs-build.yml
+```
+
+#### arguments:
+- A docs requirements file can be specified by assigning a value to
+  `DOCS_REQUIREMENTS`, which defaults to `docs-requirements.txt`.
+- The folder containing the documentation can be specified by assigning a value
+  to `DOCS_FOLDER`, which defaults to `docs`.
```

Comment:
"Folder" (as opposed to "directory") feels so Windows-y, but I think that's just me being extremely nitpicky.

Reply:
Haha, I blame @hhslepicka for that one, but honestly I have no preference, they both sound right to me even though I'm a Mac/unix guy.